### DISCUSSION TITLE
Set buildTileUrlTemplate config option for custom tile server

### DISCUSSION THREAD

**Discussion by itrnd** [radiantearth/stac-browser#457] [Q&A]
I'm using a tile server that accepts URLs of the format:
https://my-tile-server.com?collection=MyCollectionId&item=MyItemId&asset=ItemAssetKey&x=8758&y=5977&z=14
However, it's not clear to me how I should set [buildTileUrlTemplate](https://github.com/radiantearth/stac-browser/blob/main/config.js#L30) to match this format. Could you help me understand how I should pass the needed parameters? Thanks.

**Comment by itrnd** [comment]
I've tried these settings:
```
useTileLayerAsFallback: false,
buildTileUrlTemplate: ({href, asset, key, stac}) => "https://my-tile-server.com?collection={stac.collection}&item={stac.id}&asset={key}&scaling=near&x={x}&y={y}&z={z}",
displayGeoTiffByDefault: true
```
But I don't see any calls to the tile server. And the assets "Show on map" button is not present. Any idea what I'm doing wrong?

**Reply by m-mohr** [reply]
If the "Show on map" is not present, STAC Browser assumes that displaying the asset is not possible, e.g. due to a non-HTTP(S) path or a wrong media type. Can you share the asset JSON?

**Reply by chiarch84** [reply]
Thanks!
Here just 2 bands as example:
```
"B01": {
      "title": "B1",
      "type": "image/jp2",
      "roles": [
        "data"
      ],
      "gsd": 60,
      "proj:shape": [1830, 1830],
      "physicalgain_unit": "reflectance",
      "physicalgain_value": 4.09633865,
      "eo:bands": [
        {
          "name": "B01",
          "common_name": "coastal",
          "center_wavelength": 0.443,
          "full_width_half_max": 0.02
        }
      ],
      "nodata": 0,
      "data_type": "uint16",
      "scale": 10000,
      "offset": -1000,
      "href": "file:///eos/jeodpp/data/SRS/Copernicus/S2/scenes/source/L2A/2024/09/02/138/S2A_MSIL2A_20240902T130031_N0511_R138_T25TFE_20240902T180059.SAFE/GRANULE/L2A_T25TFE_A048040_20240902T130029/IMG_DATA/R60m/T25TFE_20240902T130031_B01_60m.jp2"
    },
    "B02": {
      "title": "B2",
      "type": "image/jp2",
      "roles": [
        "data"
      ],
      "gsd": 10,
      "proj:shape": [10980, 10980],
      "physicalgain_unit": "reflectance",
      "physicalgain_value": 3.75535887,
      "eo:bands": [
        {
          "name": "B02",
          "common_name": "blue",
          "center_wavelength": 0.49,
          "full_width_half_max": 0.065
        }
      ],
      "nodata": 0,
      "data_type": "uint16",
      "scale": 10000,
      "offset": -1000,
      "href": "file:///eos/jeodpp/data/SRS/Copernicus/S2/scenes/source/L2A/2024/09/02/138/S2A_MSIL2A_20240902T130031_N0511_R138_T25TFE_20240902T180059.SAFE/GRANULE/L2A_T25TFE_A048040_20240902T130029/IMG_DATA/R10m/T25TFE_20240902T130031_B02_10m.jp2"
    }
```

And the tile url is in similar to the following one (just changing the parameters):
https://jeodpp.jrc.ec.europa.eu/jiplib-view-dev/?service=DTS&collection=EO.Copernicus.S2.L2A&item=S2A_MSIL2A_20230910T100601_N0509_R022_T32TQP_20230910T161500&asset=B04&scaling=bilinear&x=8758&y=5977&z=14

**Reply by m-mohr** [reply]
Yeah, STAC Browser can't judge whether file:// paths can actually be handled by the tile server. As such it won't show the Show on Map button. So right now you need to change the STAC Browser source code to enable it.

It would probably be a good idea to have it configurable what the tile server accepts, but that's something I'd only implement if funding is available or a PR is proposed with a generic solution (e.g. another callback in config that allows to return a boolean based on the asset object or so).

**Reply by chiarch84** [reply]
Ok thank you very much with your input. Let's see what we can do. Thanks.

**Comment by chiarch84** [comment]
Dear @m-mohr could you help us here? It is probably only a matter of configuration but we have no idea how to make the button "Show on Map" appear. The configuration tried is the one that my colleague @itrnd wrote in his post. Thanks a lot for your help!
